### DISCUSSION TITLE
Welcome to Laravel Tenancy Discussions!

### DISCUSSION THREAD

**Discussion by filipprober** [filip-codes/laravel-tenancy#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by feelgood-devs** [comment]
Hi there,

first of all thanks for sharing your work :-)

It looks like you intended Tenant identification by subdomain on Registration page. But the tenant routes are pointing to         InitializeTenancyByDomain instead InitializeTenancyBySubdomain. 

Im trying to set it up on real domains but got stuck on that point. After registration the  subdomain gets opened with nothing inside.

Can you point me to the right direction?

Best Regards from munich :-)

**Reply by filipprober** [reply]
Hey @feelgood-devs,

thanks for sharing your feedback!

Yes, I've had the same scenario in a project recently. Maybe, I'll switch from InitializeTenancyByDomain to InitializeTenancyBySubdomain soon.

What you can do for now:

1. Open app/Http/Controllers/RegisteredTenantController.php and modify your `store` method:

```php
public function store(RegisterTenantRequest $request)
{
    $tenant = Tenant::create($request->only('company', 'domain'));

    $domain = strtolower($request->domain).'.'.Str::replace(['http://', 'https://'], '', config('app.url'));
    $tenant->domains()->create(['domain' => $request->domain]);

    return Inertia::location(tenant_route($domain, 'tenant.login'));
}
```

2. Open app/Http/Requests/RegisterTenantRequest.php, modify your `rules` method and remove the `prepareForValidation` method:

```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Support\Str;
use Illuminate\Validation\Rules\Password;

class RegisterTenantRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     */
    public function authorize(): bool
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
     */
    public function rules(): array
    {
        return [
            'company' => 'required|string|max:255',
            'domain' => 'required|string|max:255|lowercase|alpha_dash:ascii|unique:domains',
            'first_name' => 'required|string|max:255',
            'last_name' => 'required|string|max:255',
            'email' => 'required|string|lowercase|email|max:255',
            'password' => ['required', 'confirmed', Password::defaults()],
        ];
    }
}
```

3. Open routes/tenant.php and replace `InitializeTenancyByDomain` with `InitializeTenancyBySubdomain`:

```php
<?php

declare(strict_types=1);

use App\Http\Controllers\Tenant\ProfileController;
use Illuminate\Support\Facades\Route;
use Inertia\Inertia;
use Stancl\Tenancy\Middleware\InitializeTenancyBySubdomain;
use Stancl\Tenancy\Middleware\PreventAccessFromCentralDomains;

Route::group([
    'as' => 'tenant.',
    'middleware' => [
        'web',
        InitializeTenancyBySubdomain::class,
        PreventAccessFromCentralDomains::class,
    ],
], function () {
    Route::get('/', function () {
        return 'This is your multi-tenant application. The id of the current tenant is '.tenant('id');
    });

    Route::get('/dashboard', function () {
        return Inertia::render('Tenant/Dashboard');
    })->middleware(['auth', 'verified'])->name('dashboard');
    Route::middleware('auth')->group(function () {
        Route::get('/profile', [ProfileController::class, 'edit'])->name('profile.edit');
        Route::patch('/profile', [ProfileController::class, 'update'])->name('profile.update');
        Route::delete('/profile', [ProfileController::class, 'destroy'])->name('profile.destroy');
    });

    require __DIR__.'/auth.php';
});
```

This way, only the subdomain is stored in your database and you will be redirected to your freshly created tenant.

Please let me know, if it worked out for you!

Greets from Vienna,

Filip

**Comment by feelgood-devs** [comment]
hey filip :-)

thank you for all your efforts, but meanwhile i've found a much easier solution and tracked it down to explain how i solved it.
Cant believe thats all after so much troubleshooting! 

in file /routes/tenant.php
` 'asset_helper_tenancy' => false, `

But following diff proves that!
```
# diffs -qr --exclude=vendor --exclude=node_modules --exclude=build --exclude=storage html/ flt/
Files html/composer.lock and flt/composer.lock differ
Files html/config/tenancy.php and flt/config/tenancy.php differ
Files html/package-lock.json and flt/package-lock.json differ
Files html/public/.htaccess and flt/public/.htaccess differ
Files html/README.md and flt/README.md differ

```
only tenancy.php is germane here

```
# diffs html/config/tenancy.php flt/config/tenancy.php 
        'xyz.de',                              |         'laravel-tenancy.test',
        'asset_helper_tenancy' => false,                      |         'asset_helper_tenancy' => true,
```
There was also no need for changing to subdomain class like i've expected!
